synapse/rest/client/v2_alpha/account.py

```diff
@@ -630,34 +634,43 @@ def on_POST(self, request):
         # Try to validate as email
         if self.hs.config.threepid_behaviour_email == ThreepidBehaviour.REMOTE:
             # Ask our delegated email identity server
-            validation_session = yield self.identity_handler.threepid_from_creds(
-                self.hs.config.account_threepid_delegate_email, threepid_creds
-            )
+            try:
+                validation_session = yield self.identity_handler.threepid_from_creds(
+                    self.hs.config.account_threepid_delegate_email, threepid_creds
+                )
+            except HttpResponseException:
+                pass
         elif self.hs.config.threepid_behaviour_email == ThreepidBehaviour.LOCAL:
             # Get a validated session matching these details
             validation_session = yield self.datastore.get_threepid_validation_session(
                 "email", client_secret, sid=sid, validated=True
             )
 
-        if self._add_threepid_to_account(user_id, validation_session):
+        if validation_session and "error" not in validation_session:
```

Comment:
we should probably comment why we are checking `error` here as well as the catching the http error?